configure: error: cannot find boost/context/fcontext.hpp

**version**
` pdns-recursor-4.0.0-rc1 `

**configure**
```
'--sysconfdir=/etc/powerdns' '--enable-systemd' '--with-protobuf'
...
checking for protoc... protoc
checking for Boost headers version >= 1.42.0... yes
checking for Boost's header version... 1_55
checking for the flags needed to use pthreads... -pthread
checking for the toolset name used by Boost for g++... gcc49 -gcc
checking whether Boost is new enough to use the context library...... yes
configure: checking whether the Boost context library actually links...
checking boost/context/fcontext.hpp usability... no
checking boost/context/fcontext.hpp presence... no
checking for boost/context/fcontext.hpp... no
configure: error: cannot find boost/context/fcontext.hpp
```

**package**
```
$ dpkg -l | grep libboost
ii  libboost-context-dev:arm64     1.55.0.2                  arm64        provides a sort of cooperative multitasking on a single thread (default version)
ii  libboost-context1.55-dev:arm64 1.55.0+dfsg-3             arm64        provides a sort of cooperative multitasking on a single thread
ii  libboost-context1.55.0:arm64   1.55.0+dfsg-3             arm64        provides a sort of cooperative multitasking on a single thread
ii  libboost-dev:arm64             1.55.0.2                  arm64        Boost C++ Libraries development files (default version)
ii  libboost-iostreams1.55.0:arm64 1.55.0+dfsg-3             arm64        Boost.Iostreams Library
ii  libboost1.55-dev:arm64         1.55.0+dfsg-3             arm64        Boost C++ Libraries development files
```
**file**
```
$ sudo find / -name fcontext.hpp
/usr/include/boost/context/fcontext.hpp
```
**OS**
```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 8.5 (jessie)
Release:        8.5
Codename:       jessie
$ uname -rm
3.14.72-odroidc2 aarch64
```

Hi @ahupowerdns 

Tried the download file you provided us.
However in error.

```
checking for protoc... protoc
checking for Boost headers version >= 1.42.0... yes
checking for Boost's header version... 1_55
checking for the flags needed to use pthreads... -pthread
checking for the toolset name used by Boost for g++... gcc49 -gcc
checking whether Boost is new enough to use the context library...... yes
configure: checking whether the Boost context library actually links...
checking boost/context/fcontext.hpp usability... no
checking boost/context/fcontext.hpp presence... no
checking for boost/context/fcontext.hpp... no
configure: error: cannot find boost/context/fcontext.hpp
```
[config.zip](https://github.com/PowerDNS/pdns/files/321895/config.zip)

Hi,
same problem here with an custom build of boost(1.61) on centos 7

```
checking for Boost headers version >= 1.35.0... /opt/lib/boost/1.61-4.8//include
checking for Boost's header version... 1_61
checking for the flags needed to use pthreads... conftest.cpp: In function 'int main()':
conftest.cpp:33:24: warning: null argument where non-null required (argument 1) [-Wnonnull]
     pthread_attr_init(0); pthread_cleanup_push(0, 0);
                        ^
conftest.cpp:34:27: warning: null argument where non-null required (argument 1) [-Wnonnull]
     pthread_create(0,0,0,0); pthread_cleanup_pop(0);
                           ^
conftest.cpp:34:27: warning: null argument where non-null required (argument 3) [-Wnonnull]
conftest.cpp:32:34: warning: 'th' is used uninitialized in this function [-Wuninitialized]
 pthread_t th; pthread_join(th, 0);
                                  ^
-pthread
checking for the toolset name used by Boost for g++... gcc48 -gcc
checking whether Boost is new enough to use the context library...... yes
checking boost/system/error_code.hpp usability... yes
checking boost/system/error_code.hpp presence... yes
checking for boost/system/error_code.hpp... yes
checking for the Boost system library... yes
sed: can't read conftest.o: No such file or directory
checking boost/thread.hpp usability... yes
checking boost/thread.hpp presence... yes
checking for boost/thread.hpp... yes
checking for the Boost thread library... (cached) yes
configure: checking whether the Boost context library actually links...
checking boost/context/fcontext.hpp usability... no
checking boost/context/fcontext.hpp presence... no
checking for boost/context/fcontext.hpp... no
configure: error: cannot find boost/context/fcontext.hpp
```


Problem with boost 1.61 is, the structure of the files are changed :(
the fcontext.hpp is now found under detail/

```
/opt/lib/boost/1.61-4.8/include/boost/context
[ingo@build-el7 context]$ find .
.
./execution_context.hpp
./execution_context_v2_void.ipp
./preallocated.hpp
./stack_traits.hpp
./all.hpp
./protected_fixedsize_stack.hpp
./execution_context_v1.hpp
./flags.hpp
./pooled_fixedsize_stack.hpp
./fixedsize_stack.hpp
./stack_context.hpp
./segmented_stack.hpp
./execution_context_v2.hpp
./posix
./posix/protected_fixedsize_stack.hpp
./posix/segmented_stack.hpp
./detail
./detail/disable_overload.hpp
./detail/fcontext_sparc.hpp
./detail/fcontext_x86_64_win.hpp
./detail/fcontext_ppc.hpp
./detail/tuple.hpp
./detail/apply.hpp
./detail/exchange.hpp
./detail/index_sequence.hpp
./detail/exception.hpp
./detail/fcontext_i386.hpp
./detail/fcontext_mips.hpp
./detail/fcontext_i386_win.hpp
./detail/fcontext.hpp
./detail/fcontext_arm_win.hpp
./detail/invoke.hpp
./detail/fcontext_x86_64.hpp
./detail/fcontext_arm.hpp
./detail/fcontext_arm_mac.hpp
./detail/config.hpp
./windows
./windows/protected_fixedsize_stack.hpp
```

--- a/pdns/recursordist/configure.ac
+++ b/pdns/recursordist/configure.ac
@@ -54,7 +54,9 @@ AC_DEFUN([PDNS_SELECT_CONTEXT_IMPL], [
       LDFLAGS="$LDFLAGS $BOOST_THREAD_LDFLAGS"
     fi
     AC_MSG_NOTICE([checking whether the Boost context library actually links...])
-    BOOST_FIND_LIB([context], [$1], [boost/context/fcontext.hpp], [[]])
+    BOOST_FIND_HEADER([boost/context/fcontext.hpp], [ : ], [
+      BOOST_FIND_LIB([context], [$1], [boost/context/fcontext.hpp], [[]])
+    ])
     case $boost_cv_lib_context in
       (yes)
         AC_MSG_NOTICE([MTasker will use the Boost context library for context switching])

